templates/cluster-template-eks-clusterclass.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
```

Comment:
This isn't quite right, it doesn't even define a `Cluster` resource.

Please take a look at the [simple clusterclass template](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/templates/cluster-template-simple-clusterclass.yaml) and adapt accordingly. Note that you will need to use CC variables and patches for the template to be consumable as-is.

Reply:
CAPA seems to be a bit of a special case here. Other providers don’t include a Cluster in their ClusterClass templates, and I think it’s better to leave that for users to define. I’m happy to add it if others feel it would be helpful, but it doesn’t seem to align with common practice.

Reply:
Hmm that's a very good point. However, I still think something is wrong with this `ClusterClass`. The `ClusterClass` should be agnostic of the `Cluster` that references it, yet the names of referenced templates in this `ClusterClass` include `${CLUSTER_NAME}` - shouldn't this be `CLUSTERCLASS_NAME` so that it is independent of the `Clusters` that reference it? e.g. `spec.controlPlane.ref.name`, `spec.controlPlane.infrastructure.ref.name`, `spec.workers.machineDeployments[0].template.bootstrap.ref.name`, spec.workers.machineDeployments[0].infrastructure.ref.name` all include `${CLUSTER_NAME}`. Otherwise you end up with a `ClusterClass` per `Cluster`, which seems unnecessary - the ClusterClass should provide variables and patches that make the `ClusterClass` reusable across Clusters.